src/v/storage/kvstore.cc

```diff
@@ -608,18 +617,20 @@ void kvstore::replay_consumer::consume_records(iobuf&& records) {
     _records = std::move(records);
 }
 
-batch_consumer::stop_parser kvstore::replay_consumer::consume_batch_end() {
+ss::future<batch_consumer::stop_parser>
+kvstore::replay_consumer::consume_batch_end() {
     /*
      * build the batch and then apply all its records to the store
      */
     model::record_batch batch(
       _header, std::move(_records), model::record_batch::tag_ctor_ng{});
 
-    batch.for_each_record([this](model::record r) {
+    auto lock = co_await _store->_db_mut.get_units();
+    batch.for_each_record([this, &lock](model::record r) {
         auto key = iobuf_to_bytes(r.release_key());
         auto value = reflection::from_iobuf<std::optional<iobuf>>(
           r.release_value());
-        _store->apply_op(std::move(key), std::move(value));
+        _store->apply_op(std::move(key), std::move(value), lock);
```

Comment:
`lock` is passed in here but then not used in `apply_op` and never returned or am I missing something?

Reply:
It's a simple way to add a contract that the units have been taken / the mutex is locked.